migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,154 @@
+    mine_report_submission_id             serial                                  NOT NULL PRIMARY KEY,
+    mine_report_submission_guid           uuid                                    NOT NULL            ,
+    mine_report_id                        integer                                 NOT NULL            ,
+    received_date                         timestamp                                                   ,
+    mine_report_submission_status_code    varchar(3)                              NOT NULL            ,
+    active_ind                            boolean DEFAULT true                    NOT NULL            ,
+    create_user                           character varying(60)                   NOT NULL            ,
+    create_timestamp                      timestamp with time zone DEFAULT now()  NOT NULL            ,
+    update_user                           character varying(60)                   NOT NULL            ,
+    update_timestamp                      timestamp with time zone DEFAULT now()  NOT NULL            ,
+
+    FOREIGN KEY (mine_report_id) REFERENCES mine_report(mine_report_id) DEFERRABLE INITIALLY DEFERRED ,
+    FOREIGN KEY (mine_report_submission_status_code) REFERENCES mine_report_submission_status_code(mine_report_submission_status_code) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_submission is 'A file submission for a code required report.';
+ALTER TABLE mine_report_submission OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_document_xref
+(
+    mine_report_document_xref_guid uuid    DEFAULT gen_random_uuid() NOT NULL PRIMARY KEY,
+    mine_document_guid             uuid                              NOT NULL            ,
+    mine_report_submission_id      integer                           NOT NULL            ,
+
+    FOREIGN KEY (mine_document_guid) REFERENCES mine_document(mine_document_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_report_submission_id) REFERENCES mine_report_submission(mine_report_submission_id) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_document_xref is 'Links a mine document to report submission.';
+ALTER TABLE mine_report_document_xref OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_comment (
+    mine_report_comment_id           serial                                  NOT NULL PRIMARY KEY,
+    mine_report_comment_guid         uuid                                    NOT NULL            ,
+    mine_report_id                   integer                                 NOT NULL            ,
+    mine_report_submission_id        integer                                                     ,
+    comment                          varchar(300)                            NOT NULL            ,
+    minespace_user_id                integer                                                     ,
+    core_user_id                     integer                                                     ,
```

Comment:
re: minespace_user_id and core_user_id.... can a comment be related to both at the same time, or just one? I assume one comment record is related to one user type, and any additional commenting would be an additional record, which would be related to only one type of user? If so, we may want to constrain this in the database, to allow a single comment to be tied to either a minespace user OR a core user, but not both.

Unless I am misunderstanding this.

Reply:
I added a constraint for this table to ensure that only one of these FK's can be populated at a time.